### ISSUE TITLE
measuring quality of pre-trained your snapshot

### ISSUE THREAD

**Issue by jaesik817** [ogroth/tf-gqn#14]
Hi ogroth,

Thank you for sharing your great code.

I want to see how much your snapshot is working well for room ring camera dataset for not just ELBO and KL divergence between prior and posterior but also quality as image.

There are some options for doing that? 

Thanks.

Best Regards,

Jaesik Yoon.

**Comment by ogroth** [comment]
Hi Jaesik,
Yes, that's definitely possible. Although you need to modify the code slightly in order to do this.
The intermediate canvas can be found in the [GQN endpoints](https://github.com/ogroth/tf-gqn/blob/master/gqn/gqn_model.py#L125). You can project them into RGB images like we did in this [utility function](https://github.com/ogroth/tf-gqn/blob/master/gqn/gqn_utils.py#L134) and then expose them again and compute any metric on them which you find convenient. Additional debug metrics and summaries should be defined [here](https://github.com/ogroth/tf-gqn/blob/master/gqn/gqn_model.py#L154) in the estimator definition.
Let us know how this goes and feel free to send a pull request - this feature might be helpful for others as well. :)
Best,
Oliver

**Comment by jaesik817** [comment]
Hi Ogroth,

Thank you for your kind answer! I will try to implement that part and share the progress to you.

I have other questions as followed, could I know those things for more concisely understanding your code?

1. When you got the valid results of rooms_ring_camera, did you use the hyper parameter settings set as default in your code? (except draw #)

2. You used two GPUs, then the code can support multi-gpus option?

Thank you advance.

Best Regards,

Jaesik Yoon.

**Comment by ogroth** [comment]
Hi Jaesik,

Regarding your questions:

1. Yes, the hyper parameters in [gqn_params.py](https://github.com/ogroth/tf-gqn/blob/master/gqn/gqn_params.py) are the ones used in the rooms_ring_camera training setup.
2. We've trained on different GPUs as reported in the [README](https://github.com/ogroth/tf-gqn#notes) but we've never trained on multiple GPUs *simultaneously*, i.e. our code does not support multi-GPU training (yet).

Best,
Oliver

**Comment by jaesik817** [comment]
Hi Oliver,

Thank you for your reply.

That was really helpful to enhance my understanding for your code.

I close this issue.

Thanks.

Bests,
Jaesik.
